agent/consul/state/peering.go

```diff
@@ -212,6 +212,13 @@ func (s *Store) PeeringWrite(idx uint64, p *pbpeering.Peering) error {
 			return fmt.Errorf("cannot write to peering that is marked for deletion")
 		}
 
+		if p.State == 0 {
+			p.State = pbpeering.PeeringState_PENDING
```

Comment:
Shouldn't the caller be setting the state to a nonzero value on the re-peer?

Reply:
Hmm the caller naively does a write but I suppose it should introspect the current state a bit more